### PR TITLE
Math magicians: components project implementation

### PR DESCRIPTION
**Pull request by habasm**
# This is Components task of the math-magicians project by @habasm 
-In this project, I am continued with the development of the Math Magicians app. I also added a React component that will hold the core functionality: a calculator.
## actions
-  Used React components.
-  Used React props.
- Used React state.

image

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,29 +1,24 @@
...
+        <BrowserRouter>
+          <Calculator />
+          <Routes>
+            <Route path="/calculator" element={<Calculator />} />
+          </Routes>
+        </BrowserRouter>
+      </>
+    );
+  }
+}
```
Comment (julie-ify) [inline src/App.js:24]: - [ ] As listed in your requirements, you are to use functional components. Kindly refactor this component to a functional one instead.

File: README.md [hunk 2]
```diff
@@ -28,10 +28,10 @@
 
 <!-- PROJECT DESCRIPTION -->
```
Comment (julie-ify) [inline README.md:30]: - [ ] Kindly remove these files as they would not be needed and we want to keep our codebase clean.

  oas

File: src/components/Calculator-class-based.js [hunk 3]
```diff
@@ -0,0 +1,180 @@
+import React, { Component } from 'react';
+import { v4 as uuidv4 } from 'uuid';
+import '../scss/Calculator.scss';
+import calculate from '../logic/calculate';
+
+const buttons = [
+  {
+    symbol: 'AC',
+    id: uuidv4(),
```
Comment (julie-ify) [inline src/components/Calculator-class-based.js:9]: - [ ] I don't think we need this component. It is a duplicate of the functional component - except it is class-based. Kindly remove this component.

**Review by julie-ify — CHANGES_REQUESTED** [review]
## Hi @habasm,

Good job so far!

## Highlights!
- Good feature branch name ✔️ 
- No linter errors ✔️ 
- Descriptive pull request ✔️ 
- Descriptive commit messages ✔️ 
- Good code organization and working app ✔️
- Descriptive README ✔️ 

Hello ✋, you have done well reaching this point of the milestone 🤝 . But there are some issues that you still need to work on to go to the next project but you are almost there!

# Status: Required Changes ♻️⛔️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread and tag me `@julie-ify` if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by habasm** [comment]
## Dear @julie-ify, Thanks for comments and I am  fixed the following points
### ✔️ fixations
- unnecessary files removed 
- duplicated code removed
- codding's is manipulated as functional component instead of class component

**Review by KayLemba — APPROVED** [review]
[# STATUS: APPROVED :green_circle: :clap: 

Hello @habasm , 
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

HIGHLIGHTS:

- [x]  Your repo is professional 
- [x] Descriptive P.R
- [x] Proffessional README








 








Now you can merge this branch and move on :1st_place_medal: 
Happy Coding cheers 🥇 💯 



well-1

Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._]

**Pull request merged**